include/ocpp/common/message_queue.hpp

```diff
@@ -289,12 +314,13 @@ template <typename M> class MessageQueue {
     }
 
     MessageQueue(const std::function<bool(json message)>& send_callback, const int transaction_message_attempts,
-                 const int transaction_message_retry_interval) :
-        MessageQueue(send_callback, transaction_message_attempts, transaction_message_retry_interval, {}){};
+                 const int transaction_message_retry_interval, common::DatabaseHandlerBase& databaseHandler) :
```

Comment:
Would it be possible to pass the databaseHandler as a std::shared_ptr here? It's already a sharedptr in the charge point

Reply:
No in the chargepoint it is a unique ptr. I can make it a shared ptr if you want, then I can pass it as shared ptr of course. But actually the charge point is the 'owner' of the pointer

Reply:
Oh I totally forgot about that, then we'll have to check if the access to the database handler with a shared ptr is a good idea. unpacking the unique ptr here is definitely not recommended